src/Common/Standards/OgcApi.Net.Styles/Storage/FileSystem/StyleFileSystemStorage.cs

```diff
@@ -0,0 +1,175 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using OgcApi.Net.Resources;
+using OgcApi.Net.Styles.Model.Metadata;
+using OgcApi.Net.Styles.Model.Styles;
+using OgcApi.Net.Styles.Model.Stylesheets;
+using System.Text.Json;
+
+namespace OgcApi.Net.Styles.Storage.FileSystem;
+
+public class StyleFileSystemStorage(IOptionsMonitor<StyleFileSystemStorageOptions> options,
+    IHttpContextAccessor httpContextAccessor) : IStyleStorage
+{
+    private readonly StyleFileSystemStorageOptions _options = options.CurrentValue;
+    private readonly IHttpContextAccessor _httpContextAccessor = httpContextAccessor;
+
+    public Task<bool> StyleExists(string baseResource, string styleId)
+    {
+        var styleDirectory = Path.Combine(_options.BaseDirectory, baseResource, styleId);
+        return Task.FromResult(Directory.Exists(styleDirectory));
+    }
+
+    public Task<bool> StylesheetExists(string baseResource, string styleId, string format)
+    {
+        var stylesheetExtension = FormatToExtensionMapper.GetFileExtensionForFormat(format);
+        var stylesheetName = $"{_options.StylesheetFilename}.{format}.{stylesheetExtension}";
+        var stylesheetPath = Path.Combine(_options.BaseDirectory, baseResource, styleId, stylesheetName);
+
+        return Task.FromResult(File.Exists(stylesheetPath));
+    }
+
+    public Task<List<string>> GetAvailableFormats(string baseResource, string styleId)
+    {
+        var stylesheetsPath = Path.Combine(_options.BaseDirectory, baseResource, styleId);
+
+        if (!Directory.Exists(stylesheetsPath))
+            return Task.FromResult(new List<string>());
+
+        var stylesheets = Directory.GetFiles(stylesheetsPath);
+        var availableFormats = stylesheets
+            .Select(stylesheet => Path.GetFileName(stylesheet))
+            .Where(stylesheet => stylesheet != _options.DefaultStyleFilename && stylesheet != _options.MetadataFilename)
+            .Select(stylesheet =>
+                stylesheet
+                .Split(".")
+                .Skip(1)
+                .First())
```

Comment:
This code looks strange. If it is intended to get a file extension, then it may be better to use `Path.GetExtension`.